waddrmgr/address.go

```diff
@@ -55,6 +55,10 @@ const (
 
 	// TaprootPubKey represents a p2tr (pay-to-taproot) address type.
 	TaprootPubKey
+
+	// TaprootScript represents a p2tr (pay-to-taproot) address type that
+	// commits to a script and not just a single key.
+	TaprootScript
```

Comment:
Update godoc above to mention that it assumes usage of BIP 86? (for `TaprootPubkey`).

Reply:
I wonder if we _really_ need to distinguish the two diff pubkey types here though...

Still getting through the PR, but this seems to assume that any imported taproot addrs that have items in the script tree will never spend the keyspend path?

Reply:
Updated the godoc.

I think we should distinguish between the two, yes. The first (`TaprootPubKey`) is just what the user will use on the wallet level for receiving/sending. The second (`TaprootScript`) is what we'll use internally for things we sign over the RPC. In theory we don't really need to store those scripts in the wallet DB (we've been using witness scripts for a while now and haven't stored any of them in the wallet DB, we just derive them when needed).

So basically:
 - `TaprootPubKey`: the wallet knows the private key of the internal key and we use this through `ComputeInputScript` mainly
 - `TaprootScript`: the wallet doesn't know the private key of the internal key (we use `DeriveKey`/`DeriveNextKey`) and we use this through `SignOutputRaw` mainly. We only use this to store a script and detect incoming funds.

But maybe I'm missing a use case that we're planning to support?

Reply:
> But maybe I'm missing a use case that we're planning to support?

Nope, I think you've cleared things up nicely, thanks!